orderbook/src/database.rs

```diff
@@ -11,16 +15,145 @@ pub struct Trade {
     pub fee_amount: U256,
 }
 
-/// Abstraction over the database we use to store persistent data like orders and trade events.
-#[async_trait::async_trait]
-pub trait Database {
-    async fn block_number_of_most_recent_trade(&self) -> Result<u64>;
+// The pool uses an Arc internally.
+#[derive(Clone)]
+pub struct Database {
+    pool: PgPool,
+}
+
+impl Database {
+    pub fn new(uri: &str) -> Result<Self> {
+        Ok(Self {
+            pool: PgPool::connect_lazy(uri)?,
+        })
+    }
+
+    pub async fn block_number_of_most_recent_trade(&self) -> Result<Option<u64>> {
```

Comment:
to account for reorgs (while we are offline), do we also need to store the block_hash with each trade?

Reply:
No. The trade event updating code will always delete events starting at  `block_number_of_most_recent_trade - MAX_REORG_BLOCKS`. If any hash changes that isn't a problem because those blocks are deleted based on number.